### PR TITLE
Add test libraries

### PR DESCRIPTION
**Pull request by roliveiravictor**
# Descripción

## Disclaimer: Why add the test library here/now?

We first attempted to add the library at [Mobile Testing - Android](https://github.com/mercadolibre/fury_mobile-testing-android/pull/140/files,) expecting we would have just to bump this repository. 

Screen Shot 2022-11-30 at 10 14 34

The [linter](https://rp-ci-android.furycloud.io/blue/organizations/jenkins/mobile-testing-android/detail/mobile-testing-android/38/pipeline/) though does not allow it 'cause these libraries should be added to the allow-list. If we add the library here, it doesn't make sense to implement these libraries through core-testing. We could access them directly in the app.

In resume, it's sort of a deadlock situation.

### "We do not validate `testImplementation`, nor add test libraries in the allow-list"

The first sentence is true. The second is not. I.e.:

* [Junit](https://github.com/mercadolibre/mobile-dependencies_whitelist/blob/126b1b0060c9aaeeff02b823d88464ba6ca6777e/android-whitelist.json#L467-L469)
* [Robolectric](https://github.com/mercadolibre/mobile-dependencies_whitelist/blob/126b1b0060c9aaeeff02b823d88464ba6ca6777e/android-whitelist.json#L1563-L1580)
* [Crypto Test Utils](https://github.com/mercadolibre/mobile-dependencies_whitelist/blob/126b1b0060c9aaeeff02b823d88464ba6ca6777e/android-whitelist.json#L2054-L2056)

We know the risk of adding these libraries here. By mistake they could be added as `implementation` in the app, adding unnecessary size to the app.

## Temporary solution

While this issue is properly addressed, we would like to ask the Platform Updates team to open **another** exception so that we can add these libraries to the allow list and use them as `testImplementation` in order to avoid boilerplate on MP/ML, since we are adhering benchmark tests to both apps.

## Additional context

Here is [the test module](https://github.com/mercadolibre/fury_startup-initializer-android/pulls) we recently created, so that we could add it as `testImplementation` on both apps.

# Ticket ID
- #7557651

## En qué apps impacta mi dependencia
- [X] Mercado Libre
- [X] Mercado Pago
- [ ] SmartPOS
- [ ] Alicia: Flex / Logistics
- [ ] WMS
- [ ] Meli Store

### CODE REVIEW

File: android-whitelist.json [hunk 1]
```diff
@@ -458,6 +458,16 @@
       "name": "core",
       "version": "13\\.\\+"
     },
+    {
+      "group": "androidx\\.benchmark",
```
Comment (lcarboneti) [inline android-whitelist.json:462]: ambas son de test? Las libs de test no son validadas (debemos implementarlas bajo `testimplementation``

Comment (lcarboneti) [reply android-whitelist.json:462]: @roliveiravictor

Comment (roliveiravictor) [reply android-whitelist.json:462]: Hola Lucio, todo bien? Estoy revisando el tema con @abruno- , tenemos una inconsistencia acá sobre libs de test/allow-list ; te pido que mientras mantegas el PR abierto, porfa

**Review by caramelool — APPROVED** [review]

**Review by abruno- — APPROVED** [review]

File: android-whitelist.json (hunk 1 above)
Comment (abruno-) [reply android-whitelist.json:462]: Aca lo que no se entendio al principio es que la necesidad de agregar estas libs, es porque queremos hacer una lib de testing, por lo que las libs q se agregan van a ser usadas con implementation, y luego la lib-de-testing-nueva va a ser usada con testImplementation, como tenemos con la lib de mobile-testing

Comment (roliveiravictor) [reply android-whitelist.json:462]: Sí, ahí capaz que haga más sentido es que los módulos de prueba (o mobile-testing) tengan un linter próprio y que validen solamente implementations de libs de pruebas

**Pull request merged**